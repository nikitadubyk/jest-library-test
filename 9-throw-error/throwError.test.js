const inedibleFood = require('./throwError');

describe('Testing foods:', () => {
    function eatingWood() {
        inedibleFood('wood');
    }

    test('expect throw error', () => {
        expect(eatingWood).toThrowError(/wood/);
        expect(eatingWood).toThrowError('wood');

        expect(eatingWood).toThrowError(/^Oops, I don't eat wood$/);
    });

    test('expect throw error snapshot', () => {
        expect(eatingWood).toThrowErrorMatchingSnapshot();
    });
});
