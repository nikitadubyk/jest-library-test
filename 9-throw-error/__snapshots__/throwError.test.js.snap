// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing foods: expect throw error snapshot 1`] = `"Oops, I don't eat wood"`;
